geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/AccumuloDataStore.scala

```diff
@@ -202,9 +203,43 @@ class AccumuloDataStore(val connector: Connector,
    * @param typeName simple feature type name
    * @param sft new simple feature type
    */
-  override def updateSchema(typeName: Name, sft: SimpleFeatureType): Unit =
-    throw new UnsupportedOperationException("AccumuloDataStore does not support updateSchema")
+  override def updateSchema(typeName: Name, sft: SimpleFeatureType): Unit = {
+    import org.locationtech.geomesa.utils.geotools.RichSimpleFeatureType._
+
+    // Currently this method only allows for updating of keywords in user data. All other attempted changes will fail
+
+    // Get previous schema and user data
+    val previousSft = getSchema(typeName)
+    val schemaTypeName = sft.getTypeName()
```

Comment:
off the top of my head, i think we should be able to support changing type names for the latest schema version (8?). that might mess up geoserver layers and such though, so maybe we should be careful with it.